### PR TITLE
Optimize method for reading the files generated by the Terraform

### PR DESCRIPTION
**Pull request by baixinsui**
Fixed https://github.com/eclipse-xpanse/xpanse/issues/2068
Related to https://github.com/eclipse-xpanse/xpanse/pull/2094 and https://github.com/eclipse-xpanse/tofu-maker/pull/79

### CODE REVIEW

**Comment by luoping-12345** [comment]
LGTM

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

**Comment by profoundwu** [comment]
LGTM

File: src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java [hunk 1]
```diff
@@ -60,17 +65,31 @@ public void checkoutScripts(String workspace, TerraformScriptGitRepoDetails scri
...
-    /**
-     * Recover method for checkoutScripts.
-     *
-     * @param e GitRepoCloneException
-     */
-    @Recover
-    public void recoverCheckoutScripts(GitRepoCloneException e) {
-        log.error("Retry exhausted. Throwing exception: " + e.getMessage());
-        throw e;
+    private List<File> folderContainsScripts(String workspace,
```
Comment (swaroopar) [inline src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java:71]: its better we split this into two ->
getSourceFiles  and then pass it to validateIfFolderContainsTerraformScripts which returns void or throws exception. 
Then simply return the files returned by getSourceFiles.

File: src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java [hunk 2]
```diff
@@ -60,17 +65,31 @@ public void checkoutScripts(String workspace, TerraformScriptGitRepoDetails scri
...
+                : ""));
+        File[] files = directory.listFiles();
+        boolean isScriptsExisted = Objects.nonNull(files) && files.length > 0;
+        if (isScriptsExisted) {
+            Optional<File> tfFileOptional = Arrays.stream(files).filter(file ->
+                    file.getName().endsWith(TF_SCRIPT_FILE_EXTENSION)).findAny();
+            isScriptsExisted = tfFileOptional.isPresent();
+        }
+        if (!isScriptsExisted) {
+            throw new GitRepoCloneException(
```
Comment (swaroopar) [inline src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java:85]: Also better we throw another exception. GitRepoCloneException will not make sense here. 
If needed we can create a new custom exception.

**Review by swaroopar — CHANGES_REQUESTED** [review]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**